Enforce wildcard singleton in superuser whitelist

mpalmi opened the pull request.

mpalmi committed:
Error on wildcard if whitelist is not a singleton

In the case where set_user.superuser_whitelist has the wildcard and the wildcard
is not the only element, error and inform the user of how to fix the problem.

jconway commented on set_user.c:
```diff
@@ -318,12 +318,15 @@ set_user(PG_FUNCTION_ARGS)
 				ereport(ERROR,
 						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
-						 errmsg("Switching to superuser only allowed for privileged procedure: \'set_user_u\'")));
+						 errmsg("Switching to superuser only allowed "
```
Per the error message guidelines, this message should be more like "switching to superuser not allowed" with perhaps an errhint suggesting "use set_user_u() instead". Note that errmsg should not start with a capitalized word, nor be a complete sentence, also per the guidelines. See https://www.postgresql.org/docs/current/static/error-style-guide.html

jconway commented on set_user.c:
```diff
@@ -318,12 +318,15 @@ set_user(PG_FUNCTION_ARGS)
 				ereport(ERROR,
 						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
-						 errmsg("Switching to superuser not allowed for current user")));
+						 errmsg("Switching to superuser not allowed for current user"),
```
Similar comment here -- no capital letter S

jconway commented on set_user.c:
```diff
@@ -182,18 +183,38 @@ check_user_whitelist(Oid userId, const char *whitelist)
+		if (pg_strcasecmp(first_elem, WHITELIST_WILDCARD) == 0)
+			return true;
+	}
+
```
The entire next stanza should probably be inside"
else
{
    ....
}

jconway commented on set_user.c:
```diff
@@ -182,18 +183,38 @@ check_user_whitelist(Oid userId, const char *whitelist)
-	return false;
+	return result;
 }
 
```
I think there are at least a couple places in the README that need subtle rewording based on this.

mpalmi committed:
Clean up some set_user_u error messages

- Don't start errmsg's with capital letter
- Line-wrap some long errhint text
- Don't use complete sentences for errmsg

mpalmi commented:
@jconway feedback addressed in https://github.com/pgaudit/set_user/pull/5/commits
